dbt/adapters/glue/impl.py

```diff
@@ -534,9 +534,37 @@ def create_csv_table(self, model, agate_table):
         else:
             mode = "False"
 
-        code = f'''
-custom_glue_code_for_dbt_adapter
-csv = {json.loads(f.getvalue())}
+        csv_chunks = self._split_csv_records_into_chunks(json.loads(f.getvalue()))
+        statements = self._map_csv_chunks_to_code(csv_chunks, session, model, mode)
+        try:
+            cursor = session.cursor()
+            for statement in statements:
+                cursor.execute(statement)
+        except DbtDatabaseError as e:
+            raise DbtDatabaseError(msg="GlueCreateCsvFailed") from e
+        except Exception as e:
+            logger.error(e)
+
+    def _map_csv_chunks_to_code(self, csv_chunks: list[list[dict]], session: GlueConnection, model, mode):
+        statements = []
+        for i, csv_chunk in enumerate(csv_chunks):
+            is_first = i == 0
+            is_last = i == len(csv_chunks) - 1
+            code = "custom_glue_code_for_dbt_adapter\n"
+            if is_first:
+                code += f"""
+csv = {csv_chunk}
+"""
+            else:
+                code += f"""
+csv.extend({csv_chunk})        
+"""
+            if not is_last:
+                code += f'''
+SqlWrapper2.execute("""select 1""")
```

Comment:
What is this for?

Reply:
That's so the cursor execute can retrieve the response. Otherwise it breaks when retrieving the result on 
``` python
            if self.connection.use_arrow:
                result_bucket = self.response.get("result_bucket")
                result_key = self.response.get("result_key")
                if result_bucket and result_key:
                    pdf = get_pandas_dataframe_from_result_file(result_bucket, result_key)
                    self.result = pdf.to_dict('records')[0]

```
or on 
```
                        chunks = output.get("Data", {}).get("TextPlain", None).strip().split('\n')
                        logger.debug(f"chunks: {chunks}")
                        self.response = json.loads(chunks[0])

```

I noticed it being handled in this same way ```SqlWrapper2.execute("""select 1""")``` on other parts of ```impl.py```

Reply:
ok